collect_app/src/main/java/org/odk/collect/android/widgets/WidgetFactory.java

```diff
@@ -269,4 +285,55 @@ public static QuestionWidget createWidgetFromPrompt(FormEntryPrompt fep, Context
         return questionWidget;
     }
 
+    /**
+     * Log analytics event each time a question with a choice filter is accessed, identifying
+     * choice filters with relative expressions. This will inform communication around the fix
+     * for a long-standing bug in JavaRosa: https://github.com/opendatakit/javarosa/issues/293
+     */
+    private static void logChoiceFilterAnalytics(QuestionDef question) {
+        ItemsetBinding itemsetBinding = question.getDynamicChoices();
+
+        if (itemsetBinding != null && itemsetBinding.nodesetRef != null) {
+            if (itemsetBinding.nodesetRef.hasPredicates()) {
+                for (int level = 0; level < itemsetBinding.nodesetRef.size(); level++) {
+                    List<XPathExpression> predicates = itemsetBinding.nodesetRef.getPredicate(level);
+
+                    if (predicates != null) {
+                        // Log a hash of the form title and id joined by a space. This
+                        // will allow us to know a rough count of unique forms that use
+                        // current() in a predicate without compromising user privacy.
+                        String formIdentifier = "";
+                        FormController formController = Collect.getInstance().getFormController();
+                        if (formController != null) {
+                            String formID = formController.getFormDef().getMainInstance()
+                                    .getRoot().getAttributeValue("", "id");
+                            formIdentifier = formController.getFormTitle() + " " + formID;
+                        }
+
+                        String formIdentifierHash = FileUtils.getMd5Hash(
+                                new ByteArrayInputStream(formIdentifier.getBytes()));
+
+                        for (XPathExpression predicate : predicates) {
+                            if (predicate.toString().contains("current")) {
```

Comment:
I think this could be:
```java
Collect.getInstance().getDefaultTracker()
                                        .send(new HitBuilders.EventBuilder()
                                        .setCategory("Itemset")
                                        .setAction(predicate.toString().contains("current") ? "CurrentPredicate": "NonCurrentPredicate")
                                        .setLabel(formIdentifierHash)
                                        .build());
```

I also think @dcbriccetti would approve of this comment :)

Reply:
Can I let you in on a secret? I really am not a big inline ternary fan because I think it often hides intent. That said I understand the distaste for pretty long and mostly redundant calls in each branch. How about I pull out an `actionName` variable that I set with the ternary you suggest and use that in `setAction` so it's at least clear at a glance that there are two cases?

Reply:
Absolutely right! 

The underlying goal of the change I suggested was to separate the moving parts (the conditional expression/statement/whatever) from the fixed parts. An extracted variable `actionName` works perfectly :)

Reply:
👍